library/alloc/src/collections/vec_deque/iter.rs

```diff
@@ -12,9 +12,15 @@ use super::{count, wrap_index, RingSlices};
 /// [`iter`]: super::VecDeque::iter
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct Iter<'a, T: 'a> {
-    pub(crate) ring: &'a [T],
-    pub(crate) tail: usize,
-    pub(crate) head: usize,
+    ring: &'a [T],
+    tail: usize,
+    head: usize,
+}
+
+impl<'a, T> Iter<'a, T> {
+    pub(super) unsafe fn new(ring: &'a [T], tail: usize, head: usize) -> Self {
```

Comment:
Why is this unsafe?

Reply:
Hi @jyn514 thank for you reviewing this pull request.

I made the function unsafe because the function is effectively working with raw pointers. This was a point made by @joshtriplett on a similar pull request https://github.com/rust-lang/rust/pull/89050.

Reply:
`NonNull` is the "effectively raw pointers" bit there, not indexing. Indexing is safe.

Reply:
Removed the unsafe from the new function.